tools/genhtml.pl

```diff
@@ -155,6 +155,33 @@ ()
                                   @termination_type_value = split(/=/, $variable_value_pair[1]);
                                   @termination_id_value   = split(/=/, $variable_value_pair[2]);
                                   @corefile_value         = split(/=/, $variable_value_pair[3]);
+
+                                  # determine a category exclusively not to get duplicates - FAIL > BROK > WARN > CONF > PASS
+                             	  if ($failed_test_counter_flag == 1) {
+                             	      # ignore any BROK, WARN, or CONF event which has lower impact than FAIL
+                                      $brok_test_counter -= $brok_test_counter_flag;
+                                      $warn_test_counter -= $warn_test_counter_flag;
+                                      $conf_test_counter -= $conf_test_counter_flag;
+                             	  } elsif ($brok_test_counter_flag == 1) {
+                             	      # ignore any WARN or CONF event which has lower impact than BROK
+                                      $warn_test_counter -= $warn_test_counter_flag;
+                                      $conf_test_counter -= $conf_test_counter_flag;
+                                  } elsif ($warn_test_counter_flag == 1) {
+                                      # ignore any CONF event which has lower impact than WARN
+                                      $conf_test_counter -= $conf_test_counter_flag;
+                                  } elsif ($conf_test_counter_flag == 0) {
```

Comment:
I do not like this mess at all. We should, instead, have a global counters for fail/brok/warn/conf... and another set used inside of the loop that processes test result lines. Then we should increment the global counters once after the test output was processed, then we can easily increment only the most important result counter.